src/data/set/pointwise/smul.lean

```diff
@@ -449,6 +472,19 @@ pairwise_disjoint_image_right_iff $ λ _ _, mul_right_injective _
 
 end left_cancel_semigroup
 
+section monoid
+variables [monoid α] [mul_action α β]
+
+@[to_additive] lemma op_smul_set_smul_eq_smul_smul_set (s : set α) (a : α) (t : set β) :
+  (op a • s) • t = s • a • t :=
+by simpa using mul_smul s {a} t
```

Comment:
This strikes me as a pretty meaningless genearlization. The valuable one is the one of the form `(s : set S) (a : α) (t : set T)`, but we don't have the typeclass `op_smul_assoc_class S α T` to state it. Unless you actually intend to use this, I'd be inclined to drop it (and anything similar in this PR), perhaps with a TODO comment on the `*` version.

Reply:
I think it's important to have the statement even if it's not fully type-polymorphic. I will leave a todo.